### ISSUE TITLE
Can't parse infinity, nan and other special floating point values

### ISSUE THREAD

**Issue by Barteks2x** [falkreon/Jankson#12]
While they are nonstandard, Gson supports them, and I rely on it being possible. Gson has an option to enable parsing and outputting them.

**Comment by falkreon** [comment]
Weird, those aren't parsing? That's part of the json5 spec that we're supposed to support.

**Comment by Barteks2x** [comment]
I'm getting this error:
```
Found unrecognized token 'Infinity' while looking for a json element of any type.
Errored at line 50, column 31
```
And this for `-Infinity`:
```
Tried to parse '-' as an integer, but it appears to be invalid.
Started at line 50, column 23; Errored at line 50, column 24
```
and this for NaN:
```
Found unrecognized token 'NaN' while looking for a json element of any type.
Errored at line 684, column 58
```

**Comment by falkreon** [comment]
Oh wow, switching to the new element context broke it. Alright, that's... fixable.

**Comment by Barteks2x** [comment]
Until it's fixed, is there any workaround for this?

**Comment by falkreon** [comment]
Fixed by ef938c8 - and put into unit tests to prevent further regressions. Until the release, I recommend using the snapshot repo to pull in `1.1.3-beta.1-41`:
```groovy
repositories {
	maven {
		url = "https://repo.elytradev.com";
	}
}

dependencies {
	implementation "blue.endless:Jankson:1.1.3-beta.1-41"
}
```
